### PR TITLE
Adding support for creating Subscriptions with ramp intervals from a ramp priced plan or custom ramps

### PR DESCRIPTION
**Pull request by jorge**
We can now create subscriptions with ramp intervals in two ways. Either create a subscription and use a `plan_code` for a plan that has ramp intervals if the default ramps are desired or do the same as above while supplying a new set of ramp intervals.

### CODE REVIEW

File: src/main/java/com/ning/billing/recurly/RecurlyClient.java [hunk 1]
```diff
@@ -510,6 +510,7 @@ public void deleteAdjustment(final String adjustmentUuid) {
      * @return the newly created Subscription object on success, null otherwise
      */
     public Subscription createSubscription(final Subscription subscription) {
+        System.out.println("CLIENT createSubscription:" + subscription + "\n");
```
Comment (Smolations) [inline src/main/java/com/ning/billing/recurly/RecurlyClient.java:513]: Is this still needed?

File: src/main/java/com/ning/billing/recurly/model/RecurlyUnitCurrency.java [hunk 2]
```diff
@@ -95,6 +95,9 @@ public class RecurlyUnitCurrency {
     @XmlElement(name = "ZAR")
     private Integer unitAmountZAR;
 
+    // Pulling out of the build method to access in the rest of the class
+    private static Map amounts;
```
[the remark this answers is not in the document]
Comment (Smolations) [reply src/main/java/com/ning/billing/recurly/model/RecurlyUnitCurrency.java:99]: Also, I don't know if knowing that the property used to be in the `build` method will be valuable to devs in the future. I think perhaps the code speaks for itself here?  🤔

[the remark this answers is not in the document]
Comment (jorge) [reply src/main/java/com/ning/billing/recurly/model/RecurlyUnitCurrency.java:99]: has to be to be able to use it in a static context, the static context being the static build() method below amounts

File: src/main/java/com/ning/billing/recurly/RecurlyClient.java (hunk 1 above)
Comment (jorge) [reply src/main/java/com/ning/billing/recurly/RecurlyClient.java:513]: probably not

Comment (jorge) [reply src/main/java/com/ning/billing/recurly/RecurlyClient.java]: ya definitely not, its gone now

File: src/main/java/com/ning/billing/recurly/model/RecurlyUnitCurrency.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Smolations) [reply src/main/java/com/ning/billing/recurly/model/RecurlyUnitCurrency.java:99]: Well the `static function build()` is a factory method that returns an instance of the class, which is probably why the original `amounts` didn't need to be exposed on a class level. However, you are trying to implement it at the instance level:
```java
public Integer getUnitAmountFor(String currencyCode) {
    return RecurlyObject.integerOrNull(amounts.get(currencyCode));
}
```

The currencies and their specific amounts seem like they are unique per instance so I'm not sure how this is supposed to work with, say, set up fees? The setup fees stuff would also use this currency object for it's currencies, but if those currencies and ramp currencies use the same static `amounts` field, isn't there a possibility for a clash? 🤔

[the remark this answers is not in the document]
Comment (jorge) [reply src/main/java/com/ning/billing/recurly/model/RecurlyUnitCurrency.java]: such dynamism, wow. very java

**Review by Smolations — APPROVED** [review]

**Review by Smolations — APPROVED** [review]

**Pull request merged**